src/render.tsx

```diff
@@ -42,6 +43,8 @@ export default function render<T>(
     });
   }
 
+  detectHostComponentNamesIfNeeded();
```

Comment:
I thin we should also call it in case when `unstable_validateStringsRenderedWithinText`. This can be probably solved by moving it above the `if (unstable_validateStringsRenderedWithinText)` on L39.

Reply:
Oh absolutely, I fixed this by moving calls to `detectHostComponentNamesIfNeeded ` to queries